grpc/creds/creds.go

```diff
@@ -51,9 +63,73 @@ func (tc *transportCredentials) ClientHandshake(ctx context.Context, addr string
 	}
 }
 
+func (tc *transportCredentials) peerIsWhitelisted(peerState tls.ConnectionState) error {
+	// If there's no whitelist, all clients are OK
+	if tc.whitelist == nil {
+		return nil
+	}
+
+	// Otherwise its time to start inspecting the peer's `VerifiedChains`
+	chains := peerState.VerifiedChains
+	if len(chains) < 1 {
```

Comment:
This check is unnecessary. The loop below will happily run 0 times if this is the case.